test/Directory.Build.props

```diff
@@ -23,7 +23,7 @@
     <VSTestLogger>trx</VSTestLogger>
     <VSTestResultsDirectory>$(OutputPath)</VSTestResultsDirectory>
     <AssemblyOriginatorKeyFile>$(ToolsDir)Test.snk</AssemblyOriginatorKeyFile>
-    <CopyLocalLockFileAssemblies>false</CopyLocalLockFileAssemblies>
+    <CopyLocalLockFileAssemblies Condition="'$(TargetFramework)' == 'netcoreapp3.0'">false</CopyLocalLockFileAssemblies>
```

Comment:
I thought we set this to save on disk space. If we set this only for netcoreapp3.0 what happens to the diskspace usage on other platforms?

Reply:
Yes, we only met disk issue on dotnet core 3.0 because of below setting from build tools:

<CopyLocalLockFileAssemblies Condition="'$(CopyLocalLockFileAssemblies)' == '' and
                                        '$(TargetFrameworkIdentifier)' == '.NETStandard'">false</CopyLocalLockFileAssemblies>

<!-- Don't copy local for netcoreapp projects before 3.0 or non-exe and non-component projects. -->
<CopyLocalLockFileAssemblies Condition="'$(CopyLocalLockFileAssemblies)' == '' and
                                        '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
                                        ('$(_TargetFrameworkVersionWithoutV)' &lt; '3.0' or
                                         ('$(HasRuntimeOutput)' != 'true' and '$(EnableDynamicLoading)' != 'true'))">false</CopyLocalLockFileAssemblies>

<!-- All other project types should copy local. -->
<CopyLocalLockFileAssemblies Condition="'$(CopyLocalLockFileAssemblies)' == ''">true</CopyLocalLockFileAssemblies>


for dotnet core 3.0 packages will copy to output folder and use extra 6-7 GB. but this setting to false somehow affect test running for netfx 461 as below:
[https://dev.azure.com/dnceng/public/_build/results?buildId=528701&view=logs&j=32952595-30e7-56fa-9b86-c4579b87f5d1](https://dev.azure.com/dnceng/public/_build/results?buildId=528701&view=logs&j=32952595-30e7-56fa-9b86-c4579b87f5d1)

I'm not fully understand why this setting will affect test running on CI for netfx 461 but restrict this setting only for dotnet core 3.0 fix the disk issue also allow netfx 461 test to run on CI


---
In reply to: [382258052](https://github.com/dotnet/machinelearning/pull/4862#discussion_r382258052) [](ancestors = 382258052)